tools/list_stl_dependencies.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+#=====================================================
+# usage:
+#   cd ICEORYX_ROOT_DIR
+#   ./tools/list_stl_dependencies.sh
+#=====================================================
+
+COMPONENTS=(iceoryx_posh iceoryx_utils)
+SOURCE_DIR=(source include)
+
+
+for COMPONENT in ${COMPONENTS[@]}; do
+    for DIR in ${SOURCE_DIR[@]}; do
+        GREP_PATH="${GREP_PATH} ${COMPONENT}/$DIR"
+    done
+done
+
+echo
```

Comment:
maybe the output can be piped additionally into a file.

Reply:
Yes this is possible with the normal bash tooling. Just write
```sh
./list_stl_dependencies.sh > someFile.txt
```